language/move-bytecode-verifier/src/limits.rs

```diff
@@ -132,4 +134,40 @@ impl<'a> LimitsVerifier<'a> {
         }
         Ok(())
     }
+
+    fn verify_definitions(&self, config: &VerifierConfig) -> PartialVMResult<()> {
+        if let Some(defs) = self.resolver.function_defs() {
+            if let Some(max_function_definitions) = config.max_function_definitions {
+                if defs.len() > max_function_definitions {
+                    return Err(PartialVMError::new(
+                        StatusCode::VM_MAX_FUNTION_DEFINITIONS_REACHED,
+                    ));
+                }
+            }
+        }
+        if let Some(max_struct_definitions) = config.max_struct_definitions {
+            if let Some(defs) = self.resolver.struct_defs() {
+                if defs.len() > max_struct_definitions {
+                    return Err(PartialVMError::new(
+                        StatusCode::VM_MAX_STRUCT_DEFINITIONS_REACHED,
+                    ));
+                }
+                if let Some(max_fields_in_struct) = config.max_fields_in_struct {
```

Comment:
If someone sets `max_struct_definitions` to `None` but `max_fields_in_struct` to `Some(_)`,  the latter will be ignored (which may be confusing).

Reply:
done, thanks Sam!